met-web/src/components/engagement/listing/AdvancedSearch/SearchComponentMobile.tsx

```diff
@@ -0,0 +1,322 @@
+import React, { useState } from 'react';
+import { Grid, Stack, TextField, FormLabel, FormGroup, FormControlLabel, Checkbox } from '@mui/material';
+import { MetParagraph, MetLabel } from 'components/common';
+import { EngagementDisplayStatus } from 'constants/engagementStatus';
+import { PrimaryButton, SecondaryButton } from '../../../common';
+import dayjs from 'dayjs';
+import { formatToUTC } from 'components/common/dateHelper';
+import { SearchOptions } from './SearchTypes';
+
+interface filterParams {
+    setFilterParams: (newsearchOptions: SearchOptions) => void;
+}
+
+const AdvancedSearch: React.FC<filterParams> = ({ setFilterParams }) => {
+    const intitialStatusList = {
+        [EngagementDisplayStatus[EngagementDisplayStatus.Draft]]: false,
+        [EngagementDisplayStatus[EngagementDisplayStatus.Scheduled]]: false,
+        [EngagementDisplayStatus[EngagementDisplayStatus.Upcoming]]: false,
+        [EngagementDisplayStatus[EngagementDisplayStatus.Open]]: false,
+        [EngagementDisplayStatus[EngagementDisplayStatus.Closed]]: false,
+    };
+    const [statusFilter, setStatusFilter] = useState(() => {
+        return intitialStatusList;
+    });
+    const initialFilterParams = {
+        status_list: [],
+        created_from_date: '',
+        created_to_date: '',
+        published_from_date: '',
+        published_to_date: '',
+    };
+
+    const [selectedStatusList, setSelectedStatusList] = useState<number[]>([]);
+
+    const handleStatusFilterChange = (event: React.SyntheticEvent) => {
+        setStatusFilter({
+            ...statusFilter,
+            [(event.target as HTMLInputElement).value]: (event.target as HTMLInputElement).checked,
+        });
+        const id = (event.target as HTMLInputElement).id;
```

Comment:
I encourage you to always specify the most you can in variable name. Id of what? statusId?

Reply:
updated, thanks